dali/operators/image/color/debayer.h

```diff
@@ -0,0 +1,155 @@
+    }
+  }
+  TensorListShape<3> out_shapes(batch_size);
+  for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+    auto height = input_shapes[sample_idx][0];
+    auto width = input_shapes[sample_idx][1];
+    DALI_ENFORCE(height % 2 == 0 && width % 2 == 0,
+                 make_string("The height and width of the image to debayer must be even. However, "
+                             "the sample at idx ",
+                             sample_idx, " has shape ", input_shapes[sample_idx], "."));
+    TensorShape<3> out_shape{height, width, 3};
+    out_shapes.set_tensor_shape(sample_idx, out_shape);
+  }
+  return out_shapes;
+}
+
+inline debayer::DALIBayerPattern blue_pos2pattern(span<const int> blue_position) {
+  DALI_ENFORCE(
+      blue_position.size() == 2,
+      make_string(
+          "The ", debayer::bluePosArgName,
+          " must be specified with exactly two values to describe a position within 2D tile. Got ",
+          blue_position.size(), " values."));
+  int y = blue_position[0];
+  int x = blue_position[1];
+  int bayer_enum_val = 2 * y + x;
+  DALI_ENFORCE(
+      0 <= bayer_enum_val && bayer_enum_val <= 3,
+      make_string("The `", debayer::bluePosArgName,
+                  "` position must lie within 2x2 tile, got: ", TensorShape<2>{y, x}, "."));
+  return static_cast<debayer::DALIBayerPattern>(bayer_enum_val);
+}
+
+template <typename Backend>
+class DebayerImplBase {
+ public:
+  virtual ~DebayerImplBase() = default;
+  virtual void RunImpl(Workspace &ws) = 0;
+};
+
+}  // namespace debayer
+
+
+template <typename Backend>
+class Debayer : public SequenceOperator<Backend> {
+ public:
+  explicit Debayer(const OpSpec &spec)
+      : SequenceOperator<Backend>(spec),
+        alg_{debayer::parse_algorithm_name(spec.GetArgument<std::string>(debayer::algArgName))} {
+    if (!spec_.HasTensorArgument(debayer::bluePosArgName)) {
+      std::vector<int> blue_pos;
+      GetSingleOrRepeatedArg(spec_, blue_pos, debayer::bluePosArgName, 2);
+      static_pattern_ = debayer::blue_pos2pattern(make_span(blue_pos));
+    }
+  }
+
+ protected:
+  DISABLE_COPY_MOVE_ASSIGN(Debayer);
+  USE_OPERATOR_MEMBERS();
+
+ protected:
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) override {
+    output_desc.resize(1);
+    const auto &input_shape = ws.GetInputShape(0);
+    output_desc[0].type = ws.GetInputDataType(0);
+    output_desc[0].shape = debayer::infer_output_shape(input_shape);
+    AcquirePatternArgument(ws, input_shape.num_samples());
+    return true;
+  }
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  void AcquirePatternArgument(const Workspace &ws, int batch_size) {
+    if (!spec_.HasTensorArgument(debayer::bluePosArgName)) {
+      pattern_.resize(batch_size, static_pattern_);
+    } else {
+      const auto &tv = ws.ArgumentInput(debayer::bluePosArgName);
+      auto blue_positions_view = view<const int, 1>(tv);
```

Comment:
Just checking, I think we have something that asserts dimensionality here, so it would error out with a bit of a cryptic message, if the user provides us with for example 2D-input. And we don't strictly require 1D data, just two elements exactly?

Reply:
This is one solution if you want to be more flexible.

Reply:
done